include/godot_cpp/classes/ref.hpp

```diff
@@ -125,30 +125,33 @@ class Ref {
 			unref();
 			return;
 		}
+
 		Ref r;
 		r.reference = Object::cast_to<T>(refb);
 		ref(r);
 		r.reference = nullptr;
 	}
 
 	void operator=(const Variant &p_variant) {
-		// FIXME
+		// Needs testing, Variant has a cast to Object * here.
+
 		// Object *object = p_variant.get_validated_object();
+		Object *object = p_variant;
```

Comment:
Can't we simply check that `p_variant.type == TYPE_OBJECT`?

Reply:
No idea, lets investigate that in a new PR :) I'm not 100% sure of the variant code because I'm having other issues with variants.